catchup/networkFetcher.go

```diff
@@ -52,37 +49,37 @@ func MakeNetworkFetcher(log logging.Logger, net network.GossipNode, cfg config.L
 	return netFetcher
 }
 
-func (netFetcher *NetworkFetcher) getHTTPPeer() (network.HTTPPeer, *peerSelectorPeer, error) {
-	for ; retryCount < netFetcher.cfg.CatchupBlockDownloadRetryAttempts; retryCount++ {
+func (netFetcher *NetworkFetcher) getHTTPPeer(count int) (network.HTTPPeer, *peerSelectorPeer, int, error) {
+	for retryCount := count; retryCount < netFetcher.cfg.CatchupBlockDownloadRetryAttempts; retryCount++ {
 		psp, err := netFetcher.peerSelector.getNextPeer()
 		if err != nil {
-			if err == errPeerSelectorNoPeerPoolsAvailable {
-				netFetcher.log.Infof("FetchBlock: unable to obtain a list of peers to download the block from; will retry shortly.")
-				// this is a possible on startup, since the network package might have yet to retrieve the list of peers.
-				time.Sleep(noPeersAvailableSleepInterval)
+			if err != errPeerSelectorNoPeerPoolsAvailable {
+				err = fmt.Errorf("FetchBlock: unable to obtain a list of peers to download the block from : %w", err)
+				return nil, nil, retryCount, err
 			}
-			err = fmt.Errorf("FetchBlock: unable to obtain a list of peers to download the block from : %w", err)
-			return nil, nil, err
-		}
-		peer := psp.Peer
-		httpPeer, ok := peer.(network.HTTPPeer)
-		if ok {
-			return httpPeer, psp, nil
+			// this is a possible on startup, since the network package might have yet to retrieve the list of peers.
+			netFetcher.log.Infof("FetchBlock: unable to obtain a list of peers to download the block from; will retry shortly.")
+			time.Sleep(noPeersAvailableSleepInterval)
+		} else {
+			peer := psp.Peer
+			httpPeer, ok := peer.(network.HTTPPeer)
+			if ok {
+				return httpPeer, psp, retryCount, nil
+			}
+			netFetcher.log.Warnf("FetchBlock: non-HTTP peer was provided by the peer selector")
+			netFetcher.peerSelector.rankPeer(psp, peerRankInvalidDownload)
 		}
-		netFetcher.log.Warnf("FetchBlock: non-HTTP peer was provided by the peer selector")
-		netFetcher.peerSelector.rankPeer(psp, peerRankInvalidDownload)
 	}
-	return nil, nil, errors.New("FetchBlock: recurring non-HTTP peer was provided by the peer selector")
+	return nil, nil, netFetcher.cfg.CatchupBlockDownloadRetryAttempts, errors.New("FetchBlock: recurring non-HTTP peer was provided by the peer selector")
 }
 
 // FetchBlock function given a round number returns a block from a http peer
 func (netFetcher *NetworkFetcher) FetchBlock(ctx context.Context, round basics.Round) (*bookkeeping.Block,
 	*agreement.Certificate, time.Duration, error) {
 	// internal retry attempt to fetch the block
-	retryCount = 0
-	for ; retryCount < netFetcher.cfg.CatchupBlockDownloadRetryAttempts; retryCount++ {
+	for retryCount := 0; retryCount < netFetcher.cfg.CatchupBlockDownloadRetryAttempts; retryCount++ {
 		// keep retrying until a valid http peer is selected by the peerSelector
-		httpPeer, psp, err := netFetcher.getHTTPPeer()
+		httpPeer, psp, retryCount, err := netFetcher.getHTTPPeer(retryCount)
```

Comment:
I think it is simpler to just ignore the `retryCount` from `getHTTPPeer`. 
If you insist on using it, then you need to make sure you don't overwrite it here.

Reply:
I'll ignore the retryCount from getHTTPPeer.